distribution/lib/Standard/Snowflake/0.0.0-dev/src/Snowflake_Data_Link.enso

```diff
@@ -0,0 +1,48 @@
+from Standard.Base import all
+from Standard.Base.Enso_Cloud.Data_Link import parse_secure_value
+from Standard.Base.Enso_Cloud.Public_Utils import get_optional_field, get_required_field
+
+import Standard.Database.Connection.Connection_Options.Connection_Options
+import Standard.Database.Connection.Credentials.Credentials
+
+import project.Snowflake_Details.Snowflake_Details
+
+## PRIVATE
+type Snowflake_Data_Link
+    ## PRIVATE
+       A data-link returning a connection to the specified database.
+    Connection details:Snowflake_Details
+
+    ## PRIVATE
+       A data-link returning a query to a specific table within a database.
+    Table name:Text details:Snowflake_Details
+
+    ## PRIVATE
+    parse json -> Snowflake_Data_Link =
+        account = get_required_field "account" json expected_type=Text
+        db_name = get_required_field "database_name" json expected_type=Text
+        schema = get_optional_field "schema" json if_missing="SNOWFLAKE" expected_type=Text
+        warehouse = get_optional_field "warehouse" json if_missing="" expected_type=Text
+
+        credentials_json = get_optional_field "credentials" json
+        credentials = credentials_json.if_not_nothing <|
+            username = get_required_field "username" credentials_json |> parse_secure_value
+            password = get_required_field "password" credentials_json |> parse_secure_value
+            Credentials.Username_And_Password username password
```

Comment:
Looking at the schema, the `credentials` field does not seem optional.

Moreover, they _are not optional_ in Enso: `Snowflake_Details` has `credentials:Credentials` - not accepting `Nothing`. So `credentials` being set to `Nothing` would result in a `Type_Error`.

```suggestion
        credentials_json = get_required_field "credentials" json
        username = get_required_field "username" credentials_json |> parse_secure_value
        password = get_required_field "password" credentials_json |> parse_secure_value
        credentials = Credentials.Username_And_Password username password
```